tests/e2e/utils.js

```diff
@@ -52,6 +44,23 @@ export function getExtensionElement(id) {
   return $(`>>>[data-qa="${id}"]`);
 }
 
+export async function waitForIdleBackgroundTasks() {
+  const protocol = browser.isChromium ? 'chrome-extension' : 'moz-extension';
+
+  if (!(await browser.getUrl()).startsWith(protocol)) {
+    throw new Error(
+      'Background idle state must be checked from the extension context',
+    );
+  }
+
+  // Wait for the 'idleOptionsObservers' response
+  await browser.execute(
+    browser.isChromium
+      ? () => chrome.runtime.sendMessage({ action: 'idleOptionsObservers' })
+      : () => browser.runtime.sendMessage({ action: 'idleOptionsObservers' }),
```

Comment:
why cannot you call `chrome` on Firefox? is it still callback driven?

Reply:
There are cases where this code was executed even before the javascript code of the page is executed (as this code is ES module (deffer)). We can use `chrome`, but than is lotery, if you have "fixed" or "old" version